proxy/src/main/java/net/md_5/bungee/netty/ChannelWrapper.java

```diff
@@ -89,18 +100,27 @@ public int getEncodeVersion()
 
     public void write(Object packet)
     {
+        // ensure netty context to for less context schedules
+        // by default we are mostly in netty context anyway, but plugins can use ProxiedPlayer.unsafe().sendPacket()
+        // in non netty context
+        if ( !ch.eventLoop().inEventLoop() )
+        {
+            ch.eventLoop().execute( () -> write( packet ) );
+            return;
+        }
+
```

Comment:
Would this then be best placed in impl of ProxiedPlayer.Unsafe.sendPacket?

Reply:
if i remove it there i need to put it at multiple positions as ServerConnection.unsafe.sendPacket InitialHandler.unsafe.sendpacket and ProxiedPlayer.unsafe.sendPacket

It was just the most quick way to do it there for me because i thought there is not disadvantage of putting it ther?

Why dou you think it shouldnt be there?